### PR TITLE
N21-927 add tool status

### PR DESCRIPTION
**Pull request by arnegns**
# Short Description
Adds the tool status to tool card

## Links to Ticket and related Pull-Requests
https://ticketsystem.dbildungscloud.de/browse/N21-927

## Checklist before merging

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

File: src/store/external-tool/external-tool-display-data.ts [hunk 1]
```diff
@@ -1,6 +1,9 @@
+import { ToolConfigurationStatus } from "@/store/external-tool/tool-configuration-status.enum";
+
 export type ExternalToolDisplayData = {
 	id: string;
 	name: string;
-	logoUrl?: string;
+	logoUrl?: string | null;
```
Comment (MarvinOehlerkingCap) [inline src/store/external-tool/external-tool-display-data.ts:6]: why can it be null now?

**Review by MarvinOehlerkingCap — CHANGES_REQUESTED** [review]

File: src/store/external-tool/external-tool-display-data.ts (hunk 1 above)
Comment (arnegns) [reply src/store/external-tool/external-tool-display-data.ts]: This was a test for you :P :D

**Review by MarvinOehlerkingCap — APPROVED** [review]

File: src/composables/external-tool-mappings.composable.unit.ts [hunk 2]
```diff
@@ -390,4 +379,75 @@ describe("useExternalToolUtils", () => {
...
+
+				const status = ToolConfigurationStatus.Outdated;
+
+				return {
+					getStatusTranslationKey,
+					status,
+				};
+			};
+
+			it("should return latest translation key", () => {
```
Comment (IgorCapCoder) [inline src/composables/external-tool-mappings.composable.unit.ts:419]: should return outdated translation key

File: src/pages/rooms/external-tools/RoomExternalToolsOverview.unit.ts [hunk 3]
```diff
@@ -120,6 +121,24 @@ describe("RoomExternalToolOverview", () => {
 		});
 	});
 
+	describe("when the tools are loading", () => {
```
Comment (IgorCapCoder) [inline src/pages/rooms/external-tools/RoomExternalToolsOverview.unit.ts:124]: 👍

**Review by IgorCapCoder — CHANGES_REQUESTED** [review]

**Review by IgorCapCoder — DISMISSED** [review]

**Pull request merged**